src/hotspot/os/windows/os_windows.cpp

```diff
@@ -3160,17 +3161,22 @@ static char* map_or_reserve_memory_aligned(size_t size, size_t alignment, int fi
     // Do manual alignment
     aligned_base = align_up(extra_base, alignment);
 
-    if (file_desc != -1) {
-      os::unmap_memory(extra_base, extra_size);
-    } else {
-      os::release_memory(extra_base, extra_size);
+    bool rc = (file_desc != -1) ? os::unmap_memory(extra_base, extra_size) :
+                                  os::release_memory(extra_base, extra_size);
+    assert(rc, "release failed");
+    if (!rc) {
+      aligned_base = NULL;
+      break;
     }
 
     aligned_base = file_desc != -1 ?
       os::attempt_map_memory_to_file_at(aligned_base, size, file_desc) :
       os::attempt_reserve_memory_at(aligned_base, size);
+    attempts --;
```

Comment:
No space before unary op? E.g. `attempts--`? It might be even cleaner to move it to loop condition like `attempts-- > 0`?